File: widgetssdk/src/main/java/com/glia/widgets/core/callvisualizer/domain/VisitorCodeViewBuilderUseCase.java
package com.glia.widgets.core.callvisualizer.domain;

import android.content.Context;

import com.glia.widgets.view.VisitorCodeView;

public class VisitorCodeViewBuilderUseCase {

    private final VisitorCodeRepository visitorCodeRepository;
    private final VisitorCodeViewRepository visitorCodeViewRepository;

    public VisitorCodeViewBuilderUseCase(VisitorCodeRepository visitorCodeRepository, VisitorCodeViewRepository visitorCodeViewRepository) {
        this.visitorCodeRepository = visitorCodeRepository;
        this.visitorCodeViewRepository = visitorCodeViewRepository;
    }

    public VisitorCodeView execute(Context context) {
        VisitorCodeView visitorCodeView = visitorCodeViewRepository.createVisitorCodeView(context);
        visitorCodeView.setVisitorCodeRepository(visitorCodeRepository);
        return visitorCodeView;
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/callvisualizer/domain/VisitorCodeViewBuilderUseCase.kt
package com.glia.widgets.core.callvisualizer.domain

import android.content.Context
import com.glia.widgets.view.VisitorCodeView

internal class VisitorCodeViewBuilderUseCase {
    operator fun invoke(context: Context, closable: Boolean): VisitorCodeView = VisitorCodeView(context = context).apply { setClosable(closable) }
}
